filters/awssigner/awssigv4/sigv4_test.go

```diff
@@ -0,0 +1,359 @@
+package awssigv4
+
+import (
+	"bytes"
+	"crypto/sha256"
+	"encoding/hex"
+	"fmt"
+	"io"
+	"net/http"
+	"reflect"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/zalando/skipper/filters"
+	internal "github.com/zalando/skipper/filters/awssigner/internal"
+	"github.com/zalando/skipper/filters/filtertest"
+)
+
+func TestSignRequest(t *testing.T) {
+	var testCredentials = internal.Credentials{AccessKeyID: "AKID", SecretAccessKey: "SECRET", SessionToken: "SESSION"}
+	req, body := buildRequest("dynamodb", "us-east-1", "{}")
+	signer := NewSigner()
+	ctx := &filtertest.Context{
+		FRequest: &http.Request{},
+	}
+	optfn := func(options *SignerOptions) {
+		options.Ctx = ctx.FRequest.Context()
+	}
+	err := signer.SignHTTP(testCredentials, req, body, "dynamodb", "us-east-1", time.Unix(0, 0), optfn)
+	if err != nil {
+		t.Fatalf("expect no error, got %v", err)
+	}
+
+	expectedDate := "19700101T000000Z"
+	expectedSig := "AWS4-HMAC-SHA256 Credential=AKID/19700101/us-east-1/dynamodb/aws4_request, SignedHeaders=content-length;content-type;host;x-amz-date;x-amz-meta-other-header;x-amz-meta-other-header_with_underscore;x-amz-security-token;x-amz-target, Signature=a518299330494908a70222cec6899f6f32f297f8595f6df1776d998936652ad9"
+
+	q := req.Header
+	if e, a := expectedSig, q.Get("Authorization"); e != a {
+		t.Errorf("expect %v, got %v", e, a)
+	}
+	if e, a := expectedDate, q.Get("X-Amz-Date"); e != a {
+		t.Errorf("expect %v, got %v", e, a)
+	}
+}
+
+func TestBuildCanonicalRequest(t *testing.T) {
+	req, _ := buildRequest("dynamodb", "us-east-1", "{}")
+	req.URL.RawQuery = "Foo=z&Foo=o&Foo=m&Foo=a"
+
+	ctx := &httpSigner{
+		ServiceName:  "dynamodb",
+		Region:       "us-east-1",
+		Request:      req,
+		Time:         internal.NewSigningTime(time.Now()),
+		KeyDerivator: internal.NewSigningKeyDeriver(),
+	}
+
+	build, err := ctx.Build()
+	if err != nil {
+		t.Fatalf("expected no error, got %v", err)
+	}
+
+	expected := "https://example.org/bucket/key-._~,!@#$%^&*()?Foo=a&Foo=m&Foo=o&Foo=z"
+	if e, a := expected, build.Request.URL.String(); e != a {
+		t.Errorf("expect %v, got %v", e, a)
+	}
+}
+
+func TestSigner_SignHTTP_NoReplaceRequestBody(t *testing.T) {
```

Comment:
should this test not have bodyHash as stated in https://github.com/aws/aws-sdk-go-v2/blob/d872461174d9d665cc722432f821f98564ff0496/aws/signer/v4/v4.go#L257-L259 , so we can test this magic string?

Reply:
> should this test not have bodyHash as stated in https://github.com/aws/aws-sdk-go-v2/blob/d872461174d9d665cc722432f821f98564ff0496/aws/signer/v4/v4.go#L257-L259 , so we can test this magic string?

I copied it from [sdk](https://github.com/aws/aws-sdk-go-v2/blob/main/aws/signer/v4/v4_test.go#L206) and I believe that the test asserts that original body `{}`  remains unchanged after signing . I can add a test just for requests with no body scenarios.

Reply:
yes please add another test with http.NoBody , this one is also good to let